pkg/frontend/querymiddleware/error_caching.go

```diff
@@ -85,7 +86,7 @@ func (e *errorCachingHandler) Do(ctx context.Context, request MetricsQueryReques
 	}
 
 	e.cacheLoadAttempted.Inc()
-	key := errorCachingKey(tenant.JoinTenantIDs(tenantIDs), request)
+	key := e.keyGen.QueryRequestError(ctx, tenant.JoinTenantIDs(tenantIDs), request)
```

Comment:
I know this isn't a _change_, but since this is all relatively new, is this key going to be subject to the same cache key length issue described in #6931? Should we get ahead of it here?

Reply:
The "key" here isn't actually used directly, it's hashed and the base64 (I think) representation of the hashed key is used. The key is _also_ stored in the cached payload to handle hash collisions. I've been meaning to do something similar to resolve #6931

Reply:
> it's hashed and the base64 (I think) representation of the hashed key is used

Was the blocking comment Marco raised [here](https://github.com/grafana/mimir/pull/7043#pullrequestreview-1803908171) specifically around hashing tenant IDs in isolation (as opposed to hashing the entire key)? I'm trying to understand why his concern doesn't extend to this/the results cache.

Reply:
Yes, it's a problem when only the hashed key is used. Because this includes the full key as part of the payload there's no chance of cache collisions.